From 12ba97b9741a9f1691f2ef7417871c148dd9fa09 Mon Sep 17 00:00:00 2001
From: Ray Strode <rstrode@redhat.com>
Date: Thu, 04 Oct 2012 21:36:57 +0000
Subject: slave: only set up autologin conversations

Right now any time any session conversation is started,
we run its setup call if autologin is enabled. We should only do this
for autologin sessions, otherwise, we may repeat work done by
the greeter.

https://bugzilla.gnome.org/show_bug.cgi?id=682467
(cherry picked from commit 47c73024739a65614c6a0ddcdd1c08239798044a)
---
diff --git a/daemon/gdm-simple-slave.c b/daemon/gdm-simple-slave.c
index 59b0627..1475671 100644
--- a/daemon/gdm-simple-slave.c
+++ b/daemon/gdm-simple-slave.c
@@ -485,8 +485,12 @@ on_session_conversation_started (GdmSession       *session,
         char    *username;
         int      delay;
 
+        g_debug ("GdmSimpleSlave: session conversation started for service %s", service_name);
+
+        if (g_strcmp0 (service_name, "gdm-autologin") != 0) {
+                return;
+        }
 
-        g_debug ("GdmSimpleSlave: session conversation started");
         enabled = FALSE;
         gdm_slave_get_timed_login_details (GDM_SLAVE (slave), &enabled, &username, &delay);
         if (! enabled) {
--
cgit v0.9.0.2
